### ISSUE TITLE
Final Grades: don't display courses as failed in dashboard unless they have a final grade

### ISSUE THREAD

**Issue by pdpinch** [mitodl/micromasters#2252]
For verified (non-financial aid) courses, there is a gap between when the course ends and the certificate is created. Since we rely on the certificate in order to calculate the final grade, there can be a period of time when a learner with a passing current grade (aka progress) will see a "failed" message on the dashboard because their certificate hasn't been generated yet. 

We should modify the "failed" logic to use `final_grade.passed` and ignore the course end date.

**Comment by pdpinch** [comment]
@giocalitri does this make sense? I don't think we already have an issue for this, but I could be wrong.

**Comment by giocalitri** [comment]
If this issue is about what we discussed last night, I think it should be something like "Dashboard: allow user to pay after the end of the course until the "upgrade" deadline."

The problem is not with the final grades, but more general and the work for the MMTrack depends on this.

**Comment by pdpinch** [comment]
It's actually a different issue (which I think you had already identified). It came up today because an SCM course ended yesterday but certificates haven't been generated yet. People with passing grades are seeing "Failed"

I'll open another issue for the question of upgrading after the course end date.

**Comment by giocalitri** [comment]
OK, in the meantime bump the end of the course in the future

**Comment by pdpinch** [comment]
Is this still an issue, now that verified (no-fa) courses use the same "Progress" grade to determine passing status?

I guess we need a test where the user has a passing current grade, no final grade, and the course end date is in the past.

**Comment by pdpinch** [comment]
Yes, this is still an issue. See https://odl.zendesk.com/agent/tickets/23552
